### DISCUSSION TITLE
Foxglove Websocket IPv6 Support

### DISCUSSION THREAD

**Discussion by joewong00** [foxglove/community#709] [Q&A] [answered]
Hi

I'm currently using foxglove studio to view my ROS data and it works well. However I would like to know if foxglove websocket supports IPv6 instead of IPv4. For example, I would like to connect to the ROS master using IPv6 address through websocket connection so that it can be viewed in the browser. The format would be as follow:

ws://[ipv6-address]:8765

I have tried this but it does not work. Is it because that foxglove websocket is not yet supporting IPv6? May I know will there be any fix for this?

Thanks.

**Comment by jtbandes** [comment]
Can you please explain what you mean by “does not work”?

**Comment by joewong00** [comment]
Hi,

It says connection failed, I have already launched the foxglove bridge in the ROS master machine. The error message is to check that websocket server at ws://[ipv6_address]:8765 is reachable and supports protocol version foxglove.websocket.v1

**Reply by jtbandes** [reply]
Is the computer where you are running Studio otherwise able to reach that address?

**Reply by joewong00** [reply]
> Is the computer where you are running Studio otherwise able to reach that address?

Yes, I am able to ping to that address. I can also connect to it using the tornado websockets to visualize streaming data from the ROS master.

**Reply by jtbandes** [reply]
Are you able to use IPv6 to connect to a websocket server on your local machine?

I just tried with:

`npx @foxglove/ws-protocol-examples@latest image-server`

and was able to connect to `ws://[::1]:8765`.

<img width="1139" alt="image" src="https://github.com/foxglove/community/assets/14237/5a962216-7a9b-4dea-a8bf-cecebbf9591d">


How are you connecting using tornado websockets?

**Reply by joewong00** [reply]
Hi, I tried running on local machine, and the connection fails as well. Using ws://[::1]:8765.

**Reply by jtbandes** [reply]
This sounds like a configuration issue if your computer is not able to make any ipv6 connections. You'll probably need to search online for advice based on your specific OS, hardware, and network setup.

**Reply by bnbhat** [reply]
I faced the same issue with IPv6 address. But it got solved by setting the address param to `address:="'::'"`

**Comment by bnbhat** [comment, marked as answer]
I faced the same issue with the IPv6 address. But it got solved by setting the address param to `address:="'::'"`

for example: `ros2 launch foxglove_bridge foxglove_bridge_launch.xml address:="'::'" `
